ampersand-state.js

```diff
@@ -177,11 +178,12 @@ assign(Base.prototype, Events, {
             }
 
             isEqual = this._getCompareForType(def.type);
+            onChange = this._getOnChangeForType(def.type);
             dataType = this._dataTypes[def.type];
 
             // check type if we have one
             if (dataType && dataType.set) {
-                cast = dataType.set(newVal);
+                cast = dataType.set.call(this, newVal, options);
```

Comment:
Yeah, makes sense. I've updated this so that it doesn't bind on `set` - same as before, and will leave the bound `compare` for now.